Photo search: minor menu item text cleanup

cbracken opened the pull request:
Switches to using an elipsis character instead of three periods.

Eliminates the elipsis after 'About'. This appears to be consistent
across operating systems:

From Apple's Human Interface guidelines:
Use an ellipsis whenever choosing a menu item requires additional input
from the user. The ellipsis character (…) means a dialog or separate
window will open and prompt the user for additional information or to
make a choice.

From Microsoft's UI guidelines:
An ellipsis should be used "only when additional information is required
to perform the action. For example, the commands About, Advanced, Help,
Options, Properties, and Settings must display another window when
clicked, but don't require additional information from the user.
Therefore they don't need ellipses."

Ref: https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-anatomy/
Ref: https://docs.microsoft.com/en-us/windows/win32/uxguide/cmd-menus#using-ellipses

## Pre-launch Checklist

- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I read the [Contributors Guide].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

<!-- Links -->
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/master/CONTRIBUTING.md

domesticmouse reviewed (commented):
Holding for incoming refactoring

domesticmouse commented:
I introduced the ellipsis here because it is (was?) used to indicate that a menu item lead to a dialog box. See https://docs.microsoft.com/en-us/windows/win32/uxguide/cmd-menus#using-ellipses for details.

cbracken commented:
Ah yep - makes sense, but oddly this rule doesn't seem to apply to "About Foo" menu items. I have no idea why, but it seems to apply on Linux, Windows, and macOS (see "About this Mac" under the Apple menu, or "About Foo" under the appplication's menu).

In cases where it does apply, we should be using the elipsis character rather than three dots, and shouldn't have a space between it and the menu item.

Chrome:
![image](https://user-images.githubusercontent.com/351029/151449954-a64842d7-b1fd-432c-beb2-0818c1e831de.png)

Firefox:
![image](https://user-images.githubusercontent.com/351029/151450712-054fed91-3f30-4b70-936d-b686e250fa88.png)

gEdit:
![image](https://user-images.githubusercontent.com/351029/151450422-d530df7e-39cf-4df6-af70-42aa93f44156.png)

cbracken commented:
@domesticmouse ptal. 

Rebased and re-patched in the new code locations and updated the commit comment with the rationale from Apple's HIG and Microsoft's docs.

Sorry for the delay in getting this one updated!

domesticmouse commented:
Thanks for the update!
